code/modules/clothing/suits/reactive_armour.dm

```diff
@@ -51,91 +60,153 @@
+		if(body_part)
+			body_part.drop_limb(FALSE, TRUE)
+	playsound(get_turf(owner),'sound/machines/buzz-sigh.ogg', 50, TRUE)
+	playsound(get_turf(owner),'sound/magic/blink.ogg', 100, TRUE)
+	do_teleport(owner, get_turf(owner), tele_range, no_effects = TRUE, channel = TELEPORT_CHANNEL_BLUESPACE)
+	owner.rad_act(rad_amount)
+	reactivearmor_cooldown = world.time + reactivearmor_cooldown_duration
+	return TRUE
 
 //Fire
 
 /obj/item/clothing/suit/armor/reactive/fire
 	name = "reactive incendiary armor"
 	desc = "An experimental suit of armor with a reactive sensor array rigged to a flame emitter. For the stylish pyromaniac."
+	cooldown_message = "<span class='danger'>The reactive incendiary armor activates, but fails to send out flames as it is still recharging its flame jets!</span>"
+	emp_message = "<span class='warning'>The reactive incendiary armor's targetting system begins rebooting...</span>"
 
-/obj/item/clothing/suit/armor/reactive/fire/hit_reaction(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", final_block_chance = 0, damage = 0, attack_type = MELEE_ATTACK)
-	if(!active)
-		return FALSE
-	if(prob(hit_reaction_chance))
-		if(world.time < reactivearmor_cooldown)
-			owner.visible_message("<span class='danger'>The reactive incendiary armor on [owner] activates, but fails to send out flames as it is still recharging its flame jets!</span>")
-			return FALSE
-		owner.visible_message("<span class='danger'>[src] blocks [attack_text], sending out jets of flame!</span>")
-		playsound(get_turf(owner),'sound/magic/fireball.ogg', 100, TRUE)
-		for(var/mob/living/carbon/C in range(6, owner))
-			if(C != owner)
-				C.adjust_fire_stacks(8)
-				C.IgniteMob()
-		owner.set_fire_stacks(-20)
-		reactivearmor_cooldown = world.time + reactivearmor_cooldown_duration
-		return TRUE
+/obj/item/clothing/suit/armor/reactive/fire/reactive_activation(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", final_block_chance = 0, damage = 0, attack_type = MELEE_ATTACK)
+	owner.visible_message("<span class='danger'>[src] blocks [attack_text], sending out jets of flame!</span>")
+	playsound(get_turf(owner),'sound/magic/fireball.ogg', 100, TRUE)
+	for(var/mob/living/carbon/C in range(6, owner))
+		if(C != owner)
+			C.adjust_fire_stacks(8)
+			C.IgniteMob()
+	owner.set_fire_stacks(-20)
+	reactivearmor_cooldown = world.time + reactivearmor_cooldown_duration
+	return TRUE
+
+/obj/item/clothing/suit/armor/reactive/fire/emp_activation(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", final_block_chance = 0, damage = 0, attack_type = MELEE_ATTACK)
+	owner.visible_message("<span class='danger'>[src] just makes [attack_text] worse, by spewing molten death on [owner]!</span>")
+	playsound(get_turf(owner),'sound/magic/fireball.ogg', 100, TRUE)
+	owner.adjust_fire_stacks(12)
```

Comment:
```suggestion
	owner.adjust_fire_stacks(20)
```
reactive tele armor pops out an internal organ or pops off a limb, I think we can go for the full 20 fire stacks here (especially since the armor normally gives you -20 fire stacks).

an alternative effect for this armor could be to give you 8 fire stacks and everyone else around you -20 fire stacks, but I like the one you have here as well.